.changeset/tender-socks-trade.md

```diff
@@ -0,0 +1,5 @@
+---
+"@comet/blocks-admin": minor
+---
+
+Support `disabled` in select options when using `createCompositeBlockSelectField`
```

Comment:
Is this enabled by using `SelectFieldOption`? Is there a use case for this? We don't support dynamic options in createCompositeBlockSelectField

Reply:
> Is this enabled by using SelectFieldOption? Is there a use case for this?

Yes, using the `options` prop of `SelectField` inside `createCompositeBlockSelectField()` slightly simplifies it's code and has the side effect of allowing disabled.

I didn't think of any specific use-case but I don't see a reason to prevent it. 
I suppose this could be useful if we need to temporarily hardcode an option to be disabled for some reason, e.g. a feature is broken so we disable it until we have a good fix 🤷🏼‍♂️